File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/SearchSuggestionProvider.java
package openfoodfacts.github.scrachx.openfood.utils;

import android.content.SearchRecentSuggestionsProvider;
import openfoodfacts.github.scrachx.openfood.BuildConfig;

public class SearchSuggestionProvider extends SearchRecentSuggestionsProvider {
    public static final String AUTHORITY = BuildConfig.APPLICATION_ID+".utils.SearchSuggestionProvider";
    public static final int MODE = DATABASE_MODE_QUERIES;

    public SearchSuggestionProvider() {
        setupSuggestions(AUTHORITY, MODE);
    }
}

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/SearchSuggestionProvider.kt
package openfoodfacts.github.scrachx.openfood.utils

import android.content.SearchRecentSuggestionsProvider
import openfoodfacts.github.scrachx.openfood.BuildConfig.APPLICATION_ID

class SearchSuggestionProvider : SearchRecentSuggestionsProvider() {
    companion object {
        const val AUTHORITY = "$APPLICATION_ID.utils.SearchSuggestionProvider"
        const val MODE = DATABASE_MODE_QUERIES
    }

    init {
        setupSuggestions(AUTHORITY, MODE)
    }
}